### PR TITLE
output: add reflections

### LINKED ISSUE #154: Feature Request: Reflections (Save Crystals and Midas Gold Lara)
Now the majority of the game has been decompiled, I think it would be a good idea to port the reflection feature from TR2Main over if possible.

Specifically for the Save Crystals and Midas Gold Lara, that will make them a million times better.

### PR DESCRIPTION
**Pull request by rr-**
#### Checklist

- [X] I have read the [coding conventions](https://github.com/LostArtefacts/TR1X/blob/master/CONTRIBUTING.md#coding-conventions)
- [X] I have added a changelog entry about what my pull request accomplishes, or it is an internal change

#### Description

Resolves #154.

The tint of the save game crystals is different from the PS1 version because of how the crystal object is configured on the PC. Addressing this difference involves data changes that are beyond the scope of this pull request.

It doesn't affect the fly cheat.

Demo video: https://home.wind.garden/serve/pub/reflections.mp4

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -571,6 +571,7 @@ Not all options are turned on by default. Refer to `TR1X_ConfigTool.exe` for det
 - added contextual arrows to menu options
 - added support for animated room sprites, which also restores intended behavior in, for example, The Cistern room 0
 - added skybox support, with a default option provided for Lost Valley, Colosseum and Obelisk of Khamoon; custom level builders can use object slot `184`
+- added reflections of Midas Hand death animation and savegame crystals
```
Comment (walkawayy) [inline README.md:574]: I think `added reflections to Midas Hand death animation and savegame crystals` makes more sense

**Review by walkawayy — APPROVED** [review]
LGTM! The only things I noticed were:
1. Save crystal collision seems to be broken. I didn't have time to figure out when it broke, but I think it's a separate issue. Maybe with the bounds refactor?
2. Lara's braid doesn't turn solid with her head and continues moving. Considering the OG game never had a braid, this can probably be a separate issue for future enhancement.

**Comment by aredfan** [comment]
The crystals and midas reflections look excellent to me, they are reflecting the skyboxes even. The ingame reflections option is working as expected when enabled or disabled.

There's a new regression, the game has stopped drawing bars, rotating 3D pickups, ammo counter and ammo sprites.

**Comment by rr-** [comment]
The crystals collision bug is something we broke on develop – it's working fine in 4.3. I'll check.

**Review by aredfan — APPROVED** [review]

**Pull request merged**